### ISSUE TITLE
CP create job skills when create/edit job

### ISSUE THREAD

**Issue by patmbolger** [AgileVentures/MetPlus_tracker#629]
As a company person
When I create or edit a company job
I want to be able to create new job skills
And assign those new skills to the job

**Comment by patmbolger** [comment]
The client would like to have the ability for a company person (CP) - who is creating a job or editing an existing job - to create job skills on-the-fly and be able to add those skills to the job.

These job skills will have to be specific to the CP's company.  That is, a companies' specific job skills should not be visible to any other company.

So, with this the app will have:

1) "Global" job skills that are managed by the agency admin, and are visible to all companies, and,
2) "Company" job skills that are specific to each company.

This story includes, at a high level:

- [ ] Add model CompanySkill (this was completed by #640)
- [ ] In the Job form, add the ability for the user to add one or more skills (of type CompanySkill)
  - [ ] Explore the use of the `cocoon` gem (see similar usage, e.g. company edit where addresses can be added
  - [ ] The user should also be able to remove an added job skill - before saving the job
  - [ ] Added job skills are _not_ saved until the job is saved - and only if the job is saved successfully
  - [ ] If saving the job skill(s) fails, the job change should be rolled back and the form rendered with appropriate error messages

Note that this story does _not_ allow for the CP to edit or delete a pre-existing CompanySkill in the job form.  That will be addressed with another story - which will involve the ability to **manage** CompanySkill records.

Note also that this story does _not_ include a CompanySkill controller - that will be added with the CompanySkill **manage** story mentioned above.

**Comment by patmbolger** [comment]
I am assigning this story a size of 3.  However, we may want to break this up into 2 stories - 1) adding the model, and 2) making the changes around the form.  To be discussed with whoever is interested in tackling this.

**Comment by joaopapereira** [comment]
@patmbolger lets split this story into 2 stories as you suggested

**Comment by patmbolger** [comment]
This story is gated by #640
